### PR TITLE
ADD RESOURCE & CREATE SHARDING distsql test cases

### PR DESCRIPTION
**Pull request by isHuangXin**
ADD RESOURCE test case #19280
GSoC 2022 tasks #17939

image (1)

### CODE REVIEW

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/queryable/ConvertYamlConfigurationHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    @SneakyThrows(IOException.class)
+    private String loadExpectedRow(final String expectedFilePath) {
+        StringBuilder result = new StringBuilder();
+        String fileName = Objects.requireNonNull(ConvertYamlConfigurationHandlerTest.class.getResource(expectedFilePath)).getFile();
+        try (
+                FileReader fileReader = new FileReader(fileName);
+                BufferedReader reader = new BufferedReader(fileReader)) {
+            String line;
+            while (null != (line = reader.readLine())) {
+                if (!line.startsWith("#") && !"".equals(line.trim())) {
```
Comment (RaigorJiang) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/queryable/ConvertYamlConfigurationHandlerTest.java:129]: `"".equals(line.trim())` could be replaced to `Strings.isNullOrEmpty(line)`

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/queryable/ConvertYamlConfigurationHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+import java.util.Objects;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertFalse;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+
+public class ConvertYamlConfigurationHandlerTest extends ProxyContextRestorer {
```
Comment (RaigorJiang) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/queryable/ConvertYamlConfigurationHandlerTest.java:47]: Please declare the class as final.

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/queryable/ConvertYamlConfigurationHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    public void assertExecuteWithAddResource() throws SQLException {
+        assertExecute(resource, resourceExpectedFilePath);
+    }
+    
+    @Test
+    public void assertExecuteWithCreateSharding() throws SQLException {
+        assertExecute(sharding, shardingExpectedFilePath);
+    }
+    
+    public void assertExecute(final String feature, final String expectedFilePath) throws SQLException {
```
Comment (RaigorJiang) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/queryable/ConvertYamlConfigurationHandlerTest.java:85]: `feature` is not a good name for `resource`, maybe change to `type`?

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/ral/impl/queryable/ConvertYamlConfigurationStatementAssert.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import org.apache.shardingsphere.test.sql.parser.parameterized.asserts.SQLCaseAssertContext;
+import org.apache.shardingsphere.test.sql.parser.parameterized.jaxb.cases.domain.statement.distsql.ral.ConvertYamlConfigurationStatementTestCase;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+/**
+ * Convert yaml configuration statement assert.
+ */
+public class ConvertYamlConfigurationStatementAssert {
```
Comment (RaigorJiang) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/ral/impl/queryable/ConvertYamlConfigurationStatementAssert.java:30]: Please declare the class as final

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/ral/impl/queryable/ConvertYamlConfigurationStatementAssert.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+     * Assert convert yaml configuration statement is correct with expected parser result.
+     *
+     * @param assertContext assert context
+     * @param actual actual convert yaml configuration statement statement
+     * @param expected expected convert configuration database statement statement test case
+     */
+    public static void assertIs(final SQLCaseAssertContext assertContext, final ConvertYamlConfigurationStatement actual,
+                                final ConvertYamlConfigurationStatementTestCase expected) {
+        if (null == expected) {
+            assertNull(assertContext.getText("Actual statement should no exist."), actual);
```
Comment (RaigorJiang) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/ral/impl/queryable/ConvertYamlConfigurationStatementAssert.java:42]: Please assert `filePath` of the statement.

**Review by RaigorJiang — CHANGES_REQUESTED** [review]

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/queryable/ConvertYamlConfigurationHandlerTest.java (hunk 1 above)
Comment (isHuangXin) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/queryable/ConvertYamlConfigurationHandlerTest.java:129]: fixed~

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/queryable/ConvertYamlConfigurationHandlerTest.java (hunk 2 above)
Comment (isHuangXin) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/queryable/ConvertYamlConfigurationHandlerTest.java:47]: fixed~

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/queryable/ConvertYamlConfigurationHandlerTest.java (hunk 3 above)
Comment (isHuangXin) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/distsql/ral/queryable/ConvertYamlConfigurationHandlerTest.java:85]: fixed~

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/ral/impl/queryable/ConvertYamlConfigurationStatementAssert.java (hunk 4 above)
Comment (isHuangXin) [reply shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/ral/impl/queryable/ConvertYamlConfigurationStatementAssert.java:30]: fixed~

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/ral/impl/queryable/ConvertYamlConfigurationStatementAssert.java (hunk 5 above)
Comment (isHuangXin) [reply shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/ral/impl/queryable/ConvertYamlConfigurationStatementAssert.java:42]: fixed~

**Review by RaigorJiang — APPROVED** [review]

**Pull request merged**